### ISSUE TITLE
Get a lot of 416 "Range Not Satisfiable" Errors with the update to 4.0.4

### ISSUE THREAD

**Issue by mh4ck** [fent/node-ytdl-core#776]
Hi, 

after i updated to version 4.0.4 i directly got a lot of Status code 416 errors - after rolling back to 4.0.3 everything is working fine again.

My highWaterMark is set to 8*1024*1024 so 8MB. Making it higher or lower haven't fixed this issue.

**Comment by AMDBartek** [comment]
I am having the same issue, seems to happen randomly.

**Comment by BlazeIsClone** [comment]
yup im having some major errors as well.

**Comment by modscleo4** [comment]
Looks like #751 regressed too, I am algo getting Unable to retrieve video metadata along with HTTP 416, but more frequently than in 4.0.2

**Comment by fent** [comment]
any specific videos or formats this happens in? what options are you using?

**Comment by modscleo4** [comment]
It seems to be random
I tested my bot with this video: https://www.youtube.com/watch?v=zgezo5I3waM, but I got 416 and "Unable to retrieve video metadata" with https://www.youtube.com/watch?v=6CBp4qylX6I too

my YTDL options are
```js
filter: 'audioonly',
highWaterMark: 1 << 25,
quality: 'highestaudio',
requestOptions: {
    headers: {
        Authorization: `Bearer ${ytapikey}`,
    }
},
```

Looks like this was fixed in 4.0.5

**Comment by chaos87** [comment]
Updated to 4.0.5 and the problem occurs for every video I'm trying to download. 
I'm using options 
```
    quality: 'lowestaudio',
    audioFormat: 'mp3',
    filter: 'audioonly',
```

**Comment by galnir** [comment]
I get lots of errors on 4.0.5 aswell. Getting lots of ```Error: input stream: Video unavailable```

**Comment by chaos87** [comment]
I had to downgrade to 3.4.2 to get rid of those errors 416. Tried many combinations of filter/quality by following the README, but either I got "Output format is not available"  or error 416. 
I will upgrade once this bug is fixed.

**Comment by skick1234** [comment]
I recommended to use `4.0.3`. Almost errors will not be occurred with a valid cookies.

**Comment by Hazmi35** [comment]
I ran into the same problems on one of my servers, consistently, it seems like only music videos are affected, such as https://youtube.com/watch?v=jJPMnTXl63E, https://www.youtube.com/watch?v=dQw4w9WgXcQ, https://www.youtube.com/watch?v=SlPhMPnQ58k, and https://www.youtube.com/watch?v=QJO3ROT-A4E, on another server, I got `Error: input stream: Video unavailable` instead of this

**Comment by fent** [comment]
i wasn't able to reproduce this myself, but for those getting it, try updating to the latest version 4.0.6, which includes some fixes to requests, and see if it fixes it

**Comment by modscleo4** [comment]
I didn't get 416 anymore, but when using a lower highWaterMark, the Discord bot randomly skips some parts of the video (and no error in STDERR)

**Comment by supergrov3r** [comment]
If it helps, I'm currently using v4.0.6. Previously I was using v3.4.2 and getting the same errors (hence updating to 4.0.6).

This is what the console output looks like:
```
events.js:292
    throw er; // Unhandled 'error' event
    ^

Error: Status code: 416      at ClientRequest.<anonymous> (/workspace/node_modules/miniget/dist/index.js:156:27)      at Object.onceWrapper (events.js:422:26)      at ClientRequest.emit (events.js:315:20)      at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:596:27)      at HTTPParser.parserOnHeadersComplete (_http_common.js:119:17)      at HTTPParser.execute (<anonymous>)      at TLSSocket.socketOnData (_http_client.js:469:22)      at TLSSocket.emit (events.js:315:20)      at addChunk (_stream_readable.js:295:12)      at readableAddChunk (_stream_readable.js:271:9)
Emitted 'error' event on PassThrough instance at:
  at errorOrDestroy (internal/streams/destroy.js:108:12)
  at PassThrough.onerror (_stream_readable.js:752:7)
  at PassThrough.emit (events.js:315:20)
  at PassThrough.emit (events.js:315:20)
  at ClientRequest.<anonymous> (/workspace/node_modules/miniget/dist/index.js:161:28)
  at Object.onceWrapper (events.js:422:26)
  [... lines matching original stack trace ...]
  at HTTPParser.execute (<anonymous>)
```

**Comment by skick1234** [comment]
> i wasn't able to reproduce this myself, but for those getting it, try updating to the latest version 4.0.6, which includes some fixes to requests, and see if it fixes it

It seems to be fixed on `4.0.6`.

**Comment by chaos87** [comment]
Yes I confirm, after trying a bunch of videos that I'm not getting the 416 errors anymore after upgrading to 4.0.6
